lib/workers/repository/onboarding/branch/rebase.ts

```diff
@@ -8,26 +9,35 @@ import {
   isBranchBehindBase,
   isBranchModified,
 } from '../../../../util/git';
+import { OnboardingState, defaultConfigFile } from '../common';
 import { OnboardingCommitMessageFactory } from './commit-message';
 import { getOnboardingConfigContents } from './config';
 
-const defaultConfigFile = (config: RenovateConfig): string =>
-  configFileNames.includes(config.onboardingConfigFileName!)
-    ? config.onboardingConfigFileName!
-    : configFileNames[0];
-
 export async function rebaseOnboardingBranch(
-  config: RenovateConfig
+  config: RenovateConfig,
+  previousConfigHash?: string
 ): Promise<string | null> {
   logger.debug('Checking if onboarding branch needs rebasing');
+  const configFile = defaultConfigFile(config);
+  const existingContents =
+    (await getFile(configFile, config.onboardingBranch)) ?? '';
+  const currentConfigHash = hasha(existingContents);
+  const contents = await getOnboardingConfigContents(config, configFile);
+
+  if (is.undefined(previousConfigHash)) {
```

Comment:
why not just (!previousConfigHash) ?

Reply:
You are right, we should probably check for null as well.

Style wise I personally prefer `is` as I think it is more readable.

ill use `is.nullOrUndefined` in this case, feel free to object or suggest otherwise per best practices 😀